prime-router/src/main/kotlin/azure/SubmissionFunction.kt

```diff
@@ -27,32 +27,33 @@ class SubmissionFunction(
         val sort: String,
         val sortColumn: String,
         val cursor: OffsetDateTime?,
+        val endCursor: OffsetDateTime?,
         val pageSize: Int,
     ) {
         constructor(query: Map<String, String>) : this (
             extractSortOrder(query),
             extractSortCol(query),
-            extractCursor(query),
+            extractCursor(query, "cursor"),
+            extractCursor(query, "endCursor"),
             extractPageSize(query),
         )
 
         companion object {
             fun extractSortOrder(query: Map<String, String>): String {
-                val qSortOrder = query.getOrDefault("sort", "DESC")
-                return qSortOrder
+                return query.getOrDefault("sort", "DESC")
             }
 
             fun extractSortCol(query: Map<String, String>): String {
                 return query.getOrDefault("sortCol", "default")
             }
 
-            fun extractCursor(query: Map<String, String>): OffsetDateTime? {
-                val qResultsAfterDate = query.get("cursor")
-                return if (qResultsAfterDate != null) {
+            fun extractCursor(query: Map<String, String>, name: String): OffsetDateTime? {
+                val cursor = query.get(name)
+                return if (cursor != null) {
                     try {
-                        OffsetDateTime.parse(qResultsAfterDate)
+                        OffsetDateTime.parse(cursor)
                     } catch (e: DateTimeParseException) {
-                        throw IllegalArgumentException("cursor must be a valid datetime")
+                        throw IllegalArgumentException("$name must be a valid datetime")
```

Comment:
I like to put these variables in single quotes in case they are empty. e.g. `" '$name' must be a valid datetime"`

Reply:
Good call, just to differentiate it as a value name 👍🏼